### PR TITLE
[PW-811] Api base controller and handlers

### PR DESCRIPTION
**Pull request by florfrioli**
## Summary

- Added api base controller
- Added params handler
- Added exception handler
- Added `entity_with_code`
- Added Error response builder
- Added lib/exceptions → Ugo bases
- Added api Response Helpers
- Added Interactor Helpers → from MRP
- Added Request Performer
- Added Utility services: base - failed response mapper - request mapper - response mapper - service base helper
- Added Workers: base worker - worker helper
- Added Lambda events tracking with includes
- Added async request initializer comment

- [X] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [X] I have checked if my change needs adjustments in Active Admin
- [X] I have checked if my change needs adjustments in the API documentation
- [X] I have reviewed the files changed before releasing the pull request
- [X] I have updated the JIRA card associated with: column status, incurred hours and a comment for the QA team.
- [ ] BEFORE MERGE: I have updated the JIRA card column before merging the pull request

## Screenshots

N/A

## JIRA Card

[Link to the associated JIRA card.](https://widergy.atlassian.net/browse/PW-811)

### CODE REVIEW

File: app/services/utility_service/base.rb [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+    # def health_check
+    #   process_response(
+    #     get(@utility.health_check_url, timeout: 30),
+    #     success_response_method: :none,
+    #     failed_response_method: :none
+    #   )
+    # end
+
+    # def process_payment(params)
```
Comment (icapuccio) [inline app/services/utility_service/base.rb]: we can delete this example

File: app/services/utility_service/request_mapper.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
+module UtilityService
+  class RequestMapper
+    include PaymentGateways
+    def initialize(utility = nil)
+      @utility = utility
+    end
+
+    # def process_payment(params)
```
Comment (icapuccio) [inline app/services/utility_service/request_mapper.rb]: delete this example as well

File: app/services/utility_service/response_mapper.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    end
+
+    def default_response(_response_code, response_body)
+      {
+        message: fetch_response_message(response_body),
+        title: fetch_response_title(response_body)
+      }
+    end
+
+    def process_payment(_response_code, response_body)
```
Comment (icapuccio) [inline app/services/utility_service/response_mapper.rb]: Same

File: config/initializers/constants/api_analytics.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
+module ApiAnalytics
```
Comment (icapuccio) [inline config/initializers/constants/api_analytics.rb:1]: This file could be commented

File: lib/error_response_builder.rb [hunk 5]
```diff
@@ -0,0 +1,68 @@
+class ErrorResponseBuilder
+  attr_reader :status, :payload, :utility
+
+  # Usage:
+  #   error = ErrorResponseBuilder.new(:unprocessable_entity, utility).add_error(:invalid_token)
+  #   [error.status, error.to_h]
+  # Based on:
+  #   https://medium.com/@stevenpetryk/providing-useful-error-responses-in-a-rails-api-24c004b31a2e
```
Comment (icapuccio) [inline lib/error_response_builder.rb]: :(

**Review by ghost — APPROVED** [review]

**Pull request merged**